--- src/Terraria\Terraria.Achievements\AchievementManager.cs
+++ src/tModLoader\Terraria.Achievements\AchievementManager.cs
@@ -7,6 +_,7 @@
 using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
+using Steamworks;
 using Terraria.Social;
 using Terraria.Utilities;
 
@@ -31,7 +_,7 @@
 
 		public AchievementManager()
 		{
-			if (SocialAPI.Achievements != null)
+			if (SocialAPI.Achievements != null && SteamRemoteStorage.IsCloudEnabledForAccount() && SteamRemoteStorage.IsCloudEnabledForApp())
 			{
 				this._savePath = SocialAPI.Achievements.GetSavePath();
 				this._isCloudSave = true;

